apps/emqx/src/emqx_session.erl

```diff
@@ -412,8 +412,8 @@ enrich_delivers(ClientInfo, [D | Rest], UpgradeQoS, Session) ->
 enrich_deliver(ClientInfo, {deliver, Topic, Msg}, UpgradeQoS, Session) ->
     SubOpts =
         case Msg of
-            #message{headers = #{shared_record := SharedRecord}} ->
-                ?IMPL(Session):get_subscription(SharedRecord, Session);
+            #message{headers = #{redispatch_to := {Group, T}}} ->
```

Comment:
the tuple should be opaque out side of emqx_shared_sub, maybe move the macro to a common header file.